File: sdk/python/examples/error_handling.py
"""
CodeGate Python SDK 错误处理示例
"""

import requests
from codegate_sdk import CodeGateClient

client = CodeGateClient(
    api_key="550e8400e29b41d4a716446655440000",
    secret="a1b2c3d4e5f6...",
    project_id="550e8400e29b41d4a716446655440000"
)

try:
    # 核销激活码
    result = client.verify_code(code="ABC12345")
    
    if result['success']:
        print(f"核销成功: {result['message']}")
    else:
        error_code = result.get('error_code')
        if error_code == 'CODE_ALREADY_USED':
            print("激活码已被使用")
        elif error_code == 'CODE_NOT_FOUND':
            print("激活码不存在")
        elif error_code == 'CODE_DISABLED':
            print("激活码已禁用")
        elif error_code == 'CODE_EXPIRED':
            print("激活码已过期")
        else:
            print(f"核销失败: {result['message']}")

except requests.exceptions.HTTPError as e:
    if e.response.status_code == 401:
        print("认证失败：请检查 API Key 和 Secret")
    elif e.response.status_code == 403:
        print("权限不足：项目 ID 不匹配")
    elif e.response.status_code == 404:
        print("资源不存在")
    elif e.response.status_code == 429:
        print("请求频率过高，请稍后重试")
    else:
        print(f"请求失败: {e}")

except Exception as e:
    print(f"发生错误: {e}")

File: sdk/javascript/examples/error_handling.ts
/**
 * CodeGate SDK 错误处理示例
 *
 * 运行： npx tsx examples/error_handling.ts
 */

import { CodeGateClient, CodeGateApiError } from '../src/index';

const client = new CodeGateClient({
  apiKey: process.env.CODEGATE_API_KEY ?? 'your-api-key',
  secret: process.env.CODEGATE_SECRET ?? 'your-secret',
  projectId: process.env.CODEGATE_PROJECT_ID ?? 'your-project-id',
  baseUrl: process.env.CODEGATE_BASE_URL ?? 'https://api.example.com',
});

async function main() {
  try {
    const result = await client.verifyCode({ code: 'ABC12345' });

    if (result.success) {
      console.log('核销成功:', result.message);
    } else {
      switch (result.error_code) {
        case 'CODE_ALREADY_USED':
          console.log('激活码已被使用');
          break;
        case 'CODE_NOT_FOUND':
          console.log('激活码不存在');
          break;
        case 'CODE_DISABLED':
          console.log('激活码已禁用');
          break;
        case 'CODE_EXPIRED':
          console.log('激活码已过期');
          break;
        default:
          console.log('核销失败:', result.message);
      }
    }
  } catch (e: unknown) {
    if (e instanceof CodeGateApiError) {
      if (e.status === 401) console.log('认证失败：请检查 API Key 和 Secret');
      else if (e.status === 403) console.log('权限不足：项目 ID 不匹配');
      else if (e.status === 404) console.log('资源不存在');
      else if (e.status === 429) console.log('请求频率过高，请稍后重试');
      else console.log('API 错误:', e.status, e.detail);
    } else {
      console.log('请求失败:', (e as Error).message);
    }
  }
}

main();
